hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -241,26 +242,17 @@ public StorageStatistics provide() {
                     }
                   });
 
-      int maxThreads = conf.getInt(MAX_THREADS, DEFAULT_MAX_THREADS);
-      if (maxThreads < 2) {
-        LOG.warn(MAX_THREADS + " must be at least 2: forcing to 2.");
-        maxThreads = 2;
-      }
+      int maxThreads = getMaxThreads(conf, MAX_THREADS, DEFAULT_MAX_THREADS);
       int totalTasks = intOption(conf,
           MAX_TOTAL_TASKS, DEFAULT_MAX_TOTAL_TASKS, 1);
       long keepAliveTime = longOption(conf, KEEPALIVE_TIME,
           DEFAULT_KEEPALIVE_TIME, 0);
+
```

Comment:
ideally, keep superfluous edits to a min. I know, we all abuse that...but its good to try

Reply:
Fixed